src/services/services.ts

```diff
@@ -1246,6 +1251,86 @@ namespace ts {
+        getCompletionEntryDetails?(fileName: string, position: number, entryName: string): CompletionEntryDetails;
+        getQuickInfoAtPosition?(fileName: string, position: number): QuickInfo;
+        getNameOrDottedNameSpan?(fileName: string, startPos: number, endPos: number): TextSpan;
+        getBreakpointStatementAtPosition?(fileName: string, position: number): TextSpan;
+        getSignatureHelpItems?(fileName: string, position: number): SignatureHelpItems;
+        getRenameInfo?(fileName: string, position: number): RenameInfo;
+        findRenameLocations?(fileName: string, position: number, findInStrings: boolean, findInComments: boolean): RenameLocation[];
+        getDefinitionAtPosition?(fileName: string, position: number): DefinitionInfo[];
+        getTypeDefinitionAtPosition?(fileName: string, position: number): DefinitionInfo[];
+        getReferencesAtPosition?(fileName: string, position: number): ReferenceEntry[];
+        findReferences?(fileName: string, position: number): ReferencedSymbol[];
+        getDocumentHighlights?(fileName: string, position: number, filesToSearch: string[]): DocumentHighlights[];
+        getNavigateToItems?(searchValue: string, maxResultCount: number): NavigateToItem[];
+        getNavigationBarItems?(fileName: string): NavigationBarItem[];
+        getOutliningSpans?(fileName: string): OutliningSpan[];
+        getTodoComments?(fileName: string, descriptors: TodoCommentDescriptor[]): TodoComment[];
+        getBraceMatchingAtPosition?(fileName: string, position: number): TextSpan[];
+        getIndentationAtPosition?(fileName: string, position: number, options: EditorOptions): number;
+        getFormattingEditsForRange?(fileName: string, start: number, end: number, options: FormatCodeOptions): TextChange[];
+        getFormattingEditsForDocument?(fileName: string, options: FormatCodeOptions): TextChange[];
+        getFormattingEditsAfterKeystroke?(fileName: string, position: number, key: string, options: FormatCodeOptions): TextChange[];
+        getDocCommentTemplateAtPosition?(fileName: string, position: number): TextInsertion;
+
+        // Filters
+
+        // A plugin can implement one of the filter methods to augment, extend or modify a result
+        // prior to the host receiving it. The TypeScript language service is invoked and the
+        // result is passed to the plugin as the value of the previous parameter. If more than one
+        // plugin is registered, the plugins are consulted in the order they are returned from the
+        // program. The value passed in as previous is the result returned by the prior plugin. If a
+        // plugin returns undefined, the result passed in as previous is used and the undefined
+        // result is ignored. All plugins are consulted before the result is returned to the host.
+        // If a plugin overrides behavior of the method, no filter methods are consulted.

```

Comment:
So overrides are before (or replace) and filters are after, but overrides stop filters from running?

I'd suggest two things:
1. Document overrides and filters together, plus add an example or two to show how they interact. (Maybe even change the names.)
2. Consider making it so that overrides don't stop filters. Wouldn't it be useful for other plugins to be able to filter the output of a previous plugin?